src/settings-ui/Settings.UI/Controls/ShortcutControl/ShortcutControl.xaml.cs

```diff
@@ -113,10 +116,18 @@ private void ShortcutControl_Unloaded(object sender, RoutedEventArgs e)
             shortcutDialog.Opened -= ShortcutDialog_Opened;
             shortcutDialog.Closing -= ShortcutDialog_Closing;
 
-            App.GetSettingsWindow().Activated -= ShortcutDialog_SettingsWindow_Activated;
+            if (App.GetSettingsWindow() != null)
+            {
+                App.GetSettingsWindow().Activated -= ShortcutDialog_SettingsWindow_Activated;
+            }
 
             // Dispose the HotkeySettingsControlHook object to terminate the hook threads when the textbox is unloaded
-            hook.Dispose();
+            if (hook != null)
+            {
+                hook.Dispose();
+            }
+
+            hook = null;
```

Comment:
Doesn't this makes more sense inside the if? Why should we set it if it's already `null`.